app/jobs/modsulator_job.rb

```diff
@@ -121,6 +115,40 @@ def update_metadata(druid, xml_string, log)
     end
   end
 
+  # Open a new version for the given object if it is in the accessioned state.
+  # @param   [Dor::Item]  dor_object          The object to version
+  # @param   [String]     original_filename   The name of the uploaded file
+  # @param   [String]     user_login          The current user_login
+  # @param   [File]       log                 Log file handle
+  def version_object(dor_object, original_filename, user_login, log)
+    if(accessioned(dor_object))
+      if !DorObjectWorkflowStatus.new(dor_object.pid).can_open_version?
+        log.puts("argo.bulk_metadata.bulk_log_unable_to_version #{dor_object.pid}")  # totally unexpected
+        return
+      end
+      commit_new_version(dor_object, original_filename, user_login)
+    end
+  end
+
+  # Open a new version for the given object.
+  # @param   [Dor::Item]  dor_object          The object to version
+  # @param   [String]     original_filename   The name of the uploaded file
+  # @param   [String]     user_login          The current user_login
+  def commit_new_version(dor_object, original_filename, user_login)
+    vers_md_upd_info = {
+      :significance => 'minor',
+      :description => "Descriptive metadata upload from #{original_filename}",
+      :opening_user_name => user_login
+    }
+    dor_object.open_new_version({:vers_md_upd_info => vers_md_upd_info})
+  end
+
+  # Returns true if the given object is accessioned, false otherwise.
+  # @param   [Dor::Item]  dor_object  A DOR object
+  def accessioned(dor_object)
+    (6..8).include?(dor_object.status_info[:status_code])
```

Comment:
I always forget that Ranges have the `#cover?` method.  We should use that instead as it is more efficient.